lib/src/entity.dart

```diff
@@ -33,12 +33,12 @@ abstract class Entity extends PositionComponent {
           children: children,
           priority: priority,
         ) {
-    if (behaviors != null) {
-      addAll(behaviors);
-    }
-
     this.children.register<Behavior>();
     _behaviors = this.children.query<Behavior>();
+
+    if (behaviors != null) {
+      Future.wait<void>(behaviors.map(addBehavior));
```

Comment:
Shouldn't we use a `forEach` here instead? We not using the returned value at all.

Reply:
This would ensure the order of behaviors is added as expected because of asyncness. Enforcing both the rule and what the user wanted to add. This is technically a simplified `addAll`